pkg/sr/api.go

```diff
@@ -348,42 +348,59 @@ func (cl *Client) Schemas(ctx context.Context, subject string) ([]SubjectSchema,
 	return schemas, firstErr
 }
 
-// CreateSchema attempts to create a schema in the given subject.
+// CreateSchemaID attempts to create a schema with a fixed ID and version ID in the given subject,
+// and returns the globally unique identifier assigned by the registry. If the id is set to -1 or 0,
+// the registry will try to allocate a new identifier or reuse an existing one (see below).
+// If the versionID is set to -1 or 0, it will be omitted when creating the schema.
 //
-// This supports param [Normalize].
-func (cl *Client) CreateSchema(ctx context.Context, subject string, s Schema) (SubjectSchema, error) {
-	return cl.CreateSchemaWithIDAndVersion(ctx, subject, s, -1, -1)
-}
-
-// CreateSchemaWithIDAndVersion attempts to create a schema with a fixed ID and
-// version ID in the given subject. If the id is set to -1 or 0, this method is
-// equivalent to CreateSchema(). If the versionID is set to -1 or 0, it will be
-// omitted when creating the schema.
+// If the same schema is registered under a different subject, the same identifier will be returned.
+// If the schema is identical to an existing schema in the registry, a new version will not be created.
+// Instead, the registry will return the identifier of the existing schema.
 //
 // This supports param [Normalize].
-func (cl *Client) CreateSchemaWithIDAndVersion(ctx context.Context, subject string, s Schema, id, versionID int) (SubjectSchema, error) {
+func (cl *Client) CreateSchemaID(ctx context.Context, subject string, s Schema, id, versionID int) (int, error) {
 	// POST /subjects/{subject}/versions => returns ID
-	// Newer SR returns the full SubjectSchema, but old does not, so we
-	// re-request to find the full information.
 	path := pathSubjectWithVersion(subject)
 	var into struct {
 		ID int `json:"id"`
 	}
 	if id == -1 {
 		if err := cl.post(ctx, path, s, &into); err != nil {
-			return SubjectSchema{}, err
+			return 0, err
 		}
 	} else {
 		ss := SubjectSchema{Schema: s, ID: id}
 		if versionID != -1 {
 			ss.Version = versionID
 		}
 		if err := cl.post(ctx, path, ss, &into); err != nil {
-			return SubjectSchema{}, err
+			return 0, err
 		}
 	}
+	return into.ID, nil
+}
 
-	usages, err := cl.SchemaUsagesByID(ctx, into.ID)
+// CreateSchema attempts to create a schema in the given subject.
+//
+// This supports param [Normalize].
+func (cl *Client) CreateSchema(ctx context.Context, subject string, s Schema) (SubjectSchema, error) {
+	return cl.CreateSchemaWithIDAndVersion(ctx, subject, s, -1, -1)
+}
+
+// CreateSchemaWithIDAndVersion attempts to create a schema with a fixed ID and
+// version ID in the given subject. If the id is set to -1 or 0, this method is
```

Comment:
Sorry to add comments to not your code but... while you're in here, can you change this to `If the id is <= 0`, change `versionID` to version, and fixup the next line as well? Also, can you change `CreateSchema()` to `[CreateSchema]`?

Reply:
No problem adding more changes to this PR that are not directly related. Happy to help.